python/tests/test_unbiased.py

```diff
@@ -0,0 +1,501 @@
+@ pytest.mark.parametrize("gap", [1e-13, -1e-13, -0.5])
+@ pytest.mark.parametrize("q_deg", [1, 2, 3])
+def test_unbiased_rhs(ct, gap, q_deg):
+
+    # set quadrature degree
+    quadrature_degree = q_deg
+
+    # Compute lame parameters
+    plane_strain = False
+    E = 1e3
+    nu = 0.1
+    mu_func, lambda_func = lame_parameters(plane_strain)
+    mu = mu_func(E, nu)
+    lmbda = lambda_func(E, nu)
+    sigma = sigma_func(mu, lmbda)
+
+    # Nitche parameters and variables
+    theta = 1
+    gamma = 10
+
+    # create meshes and function spaces
+    V_ufl, V_cuas = create_functionspaces(ct, gap)
+    mesh_ufl = V_ufl.mesh
+    mesh_cuas = V_cuas.mesh
+    tdim = mesh_ufl.topology.dim
+    gdim = mesh_ufl.geometry.dim
+
+    def _u0(x):
+        values = np.zeros((gdim, x.shape[1]))
+        for i in range(tdim):
+            for j in range(x.shape[1]):
+                values[i, j] = np.sin(x[i, j])
+        return values
+
+    # DG ufl 'contact'
+    u0 = _fem.Function(V_ufl)
+    u0.interpolate(_u0)
+    v0 = ufl.TestFunction(V_ufl)
+    metadata = {"quadrature_degree": quadrature_degree}
+    dS = ufl.Measure("dS", domain=mesh_ufl, metadata=metadata)
+
+    n = ufl.FacetNormal(mesh_ufl)
+
+    # Scaled Nitsche parameter
+    h = ufl.CellDiameter(mesh_ufl)
+    gamma_scaled = gamma * E
+
+    # Contact terms formulated using ufl
+    F = -0.5 * 1 / (gamma_scaled / h('+')) * R_minus(ufl.dot(sigma(u0('+')) * n('+'), n('+'))
+                                                     + (gamma_scaled / h('+'))
+                                                     * (gap - ufl.dot(u0('-') - u0('+'), n('+'))))\
+        * (theta * ufl.dot(sigma(v0('+')) * n('+'), n('-'))
+           - (gamma_scaled / h('+')) * ufl.dot(v0('+') - v0('-'), n('-'))) * \
+        dS
+
+    F += -0.5 * 1 / (gamma_scaled / h('-')) * R_minus(ufl.dot(sigma(u0('-')) * n('-'), n('-'))
+                                                      + (gamma_scaled / h('-'))
+                                                      * (gap - ufl.dot(u0('+') - u0('-'), n('-')))) \
+        * (theta * ufl.dot(sigma(v0('-')) * n('-'), n('+'))
+           - (gamma_scaled / h('-')) * ufl.dot(v0('-') - v0('+'), n('+'))) * \
+        dS
+    F = _fem.form(F)
+    b = _fem.petsc.create_vector(F)
+    b.zeroEntries()
+    _fem.petsc.assemble_vector(b, F)
+
+    # Custom assembly
+    cells, facets_cg = locate_contact_facets_cuas(V_cuas, gap)
+
+    # fem functions
+
+    def _u1(x):
+        values = np.zeros((gdim, x.shape[1]))
+        for i in range(tdim):
+            for j in range(x.shape[1]):
+                if i == tdim - 1:
+                    values[i, j] = np.sin(x[i, j] + gap)
+                else:
+                    values[i, j] = np.sin(x[i, j])
+        return values
+
+    u1 = _fem.Function(V_cuas)
+    v1 = ufl.TestFunction(V_cuas)
+
+    u1.interpolate(_u0, cells[0])
+    u1.interpolate(_u1, cells[1])
+
```

Comment:
```suggestion
u1.x.scatter_forward()
```
To make sure this test would work in parallel (even if we will never run it in parallel).